packages/@react-stately/datepicker/src/useDatePickerState.ts

```diff
@@ -113,8 +113,14 @@ export function useDatePickerState<T extends DateValue = DateValue>(props: DateP
   };
 
   let selectTime = (newValue: TimeValue) => {
```

Comment:
there are a lot of different ways i could go about writing this conditional so i was wondering if there was a particular way that we'd like to do this?